src/modules/Speech_Recognition/Whisper.py

```diff
@@ -63,6 +80,59 @@ def transcribe_with_whisper(
     keep_numbers: bool = False,
 ) -> TranscriptionResult:
     """Transcribe with whisper"""
+    # Info: Monkey Patch FasterWhisperPipeline.detect_language to include error handling for low confidence
+    src = textwrap.dedent(inspect.getsource(whisperx.asr.FasterWhisperPipeline.detect_language))
+    # Replace the relevant part of the method
+    start_token = "if audio.shape[0] < N_SAMPLES:"
+    end_token = "return language"
+    replacement = """\
+    #Added imports
+    from modules.console_colors import ULTRASINGER_HEAD, blue_highlighted, red_highlighted
+    from Settings import Settings
+    from inputimeout import inputimeout, TimeoutOccurred
+    #End Import addition
+    if audio.shape[0] < N_SAMPLES:
+        print("Warning: audio is shorter than 30s, language detection may be inaccurate.")
+    model_n_mels = self.model.feat_kwargs.get("feature_size")
+    segment = log_mel_spectrogram(audio[: N_SAMPLES],
+                                    n_mels=model_n_mels if model_n_mels is not None else 80,
+                                    padding=0 if audio.shape[0] >= N_SAMPLES else N_SAMPLES - audio.shape[0])
+    encoder_output = self.model.encode(segment)
+    results = self.model.model.detect_language(encoder_output)
+    language_token, language_probability = results[0][0]
+    language = language_token[2:-2]
+    print(f"Detected language: {language} ({language_probability:.2f}) in first 30s of audio...")
+    #Added handling for low detection probability
+    if language_probability < Settings.CONFIDENCE_THRESHOLD:
+        print(f"{ULTRASINGER_HEAD} {red_highlighted('Warning:')} Language detection probability for detected language {language} is below {Settings.CONFIDENCE_THRESHOLD}, results may be inaccurate.")
+        print(f"{ULTRASINGER_HEAD} Override the language below or re-run with parameter {blue_highlighted('--language xx')} to specify the song language...")    
+        try:  
+            response = inputimeout(  
+                prompt=f"{ULTRASINGER_HEAD} Do you want to continue with {language} (default) or override with another language (y)? (y/n): ",  
+                timeout=Settings.CONFIDENCE_PROMPT_TIMEOUT  
+            ).strip().lower()  
+        except TimeoutOccurred:
+            import locale
+            print(f"{ULTRASINGER_HEAD} No user input received in {Settings.CONFIDENCE_PROMPT_TIMEOUT} seconds. Attempting automatic override with system locale.")
+            print(f"{ULTRASINGER_HEAD} Trying to get language from default locale...")  
+            current_locale = locale.getlocale()
+            if current_locale[0]:  
+                language_code = current_locale[0][:2].strip().lower()
+                print(f"{ULTRASINGER_HEAD} Found language code: {language_code} in locale. Setting language to {blue_highlighted(language_code)}...")
```

Comment:
What we also could us is the language info from yt (I have an upcoming branch to get this info).  And also we could use Musicbrainz. Just to infrom you :)